R/http.R

```diff
@@ -23,30 +23,44 @@ httpRequest <- function(
   headers <- c(headers, authHeaders(authInfo, method, path), httpHeaders())
   certificate <- requestCertificate(service$protocol, authInfo$certificate)
 
-  # perform request
-  httpResponse <- httpLibCurl(
-    protocol = service$protocol,
-    host = service$host,
-    port = service$port,
-    method = method,
-    path = path,
-    headers = headers,
-    timeout = timeout,
-    certificate = certificate
-  )
-
-  while (isRedirect(httpResponse$status)) {
-    service <- redirectService(service, httpResponse$location)
+  if (isTRUE(getOption("rsconnect.httr2", TRUE))) {
```

Comment:
Any reason to use a new option name rather than `rsconnect.http`, which existed until https://github.com/rstudio/rsconnect/pull/1282?

```r
options(rsconnect.http = "httr2")
```

An argument for a new option is that `rsconnect.http` was deprecated (and thus removed).

That said, there are a number of `rsconnect.http.*` options which remain. I don't have a strong opinion, here.